# Cascade Refrigeration System Project

Date: 7 Dec 2024

- Name: Ma, Xianlong
- Email: marco050601@berkeley.edu


## Summary and Problem Statement

Due to the high global warming potential of synthetic refrigerants, many of them are going to be outlawed. Therefore, there is a shift to using natural refrigerants such as carbon dioxide and ammonia. One common industrial freezer system is a CO2-ammonia cascade system. Here, the CO2 operates on the cold side and ammonia on the hot side. The freezer must operate at -20 C. 

If installing this system in the East bay, what will be the optimum temperatures for CO2 in the condenser and ammonia in the evaporator? Also propose an industrial application along with its cooling capacity.

-------

## Background information

A cascade refrigeration system is a two-stage system used to achieve very low temperatures that cannot be practically or efficiently reached by a single refrigeration cycle. It operates using two or more separate refrigeration cycles, each employing a different refrigerant suited to its specific temperature range and distribute the workload across two cycles. These cycles are thermally connected through a cascade heat exchanger where heat is transferred between the high-temperature and low-temperature cycles.

#### Cascade Refrigeration Systems: These are used for applications requiring very low temperatures (e.g., industrial freezers). In a cascade system:
- **Cycle:** The low-temperature cycle uses CO2 as the refrigerant, operating below -20°C.
The high-temperature cycle uses ammonia, which rejects heat to the environment at higher temperatures.

#### Refrigerants:
- **CO2:** High volumetric cooling capacity, environmentally friendly, operates efficiently at low temperatures.
- **Ammonia:** High energy efficiency, zero global warming potential (GWP), commonly used in industrial systems.
- Natural refrigerants like CO2 and ammonia have low or zero global warming potential (GWP) and ozone depletion potential (ODP), making cascade systems environmentally sustainable.

#### Operating Conditions:
- **Freezer temperature:** -20°C (evaporator temperature of the CO2​ cycle).
- **Ambient temperature:** Typical East Bay temperatures range from 10°C to 25°C, influencing the condenser temperature for ammonia.

-------

## Modeling the Thermodynamic Process



In [1]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.interpolate import interp1d
!pip3 install sympy
import sympy as sp
!pip3 install CoolProp
from CoolProp.CoolProp import PropsSI

## Low-Temperature Cycle (CO2 Cycle)

**1-2 Isentropic Compression:** CO2 in vapor form is compressed to a higher pressure, increasing its temperature. The compression process prepares the refrigerant for heat rejection in the cascade heat exchanger.

**2-3 Isobaric Heat Rejection:** The high-pressure, high-temperature CO2 vapor enters the cascade heat exchanger, where it releases heat to the ammonia refrigerant from the high-temperature cycle. This step cools the CO2 refrigerant and condenses it into a high-pressure liquid.

**3-4 Isenthalpic Expansion:** The high-pressure CO2 liquid passes through an expansion valve, where its pressure and temperature drop significantly. This prepares the refrigerant for absorbing heat in the evaporator.

**4-1 Isobaric Heat Absorption:** The low-pressure, low-temperature CO2 refrigerant absorbs heat from the freezer space or product being cooled. This is the primary cooling effect of the low-temperature cycle.

Here are the properties for cycle B, CO2 cycle:
Note assuming:
- P6=P7 is the condensation pressure of ammonia in the condenser.
- It corresponds to the saturation pressure of ammonia at the condenser temperature $$\text{T}_\text{condensation}$$
  $$\text{T}_\text{condensation} =  \text{T}_\text{ambient} + \text{T}_\text{approach}$$



In [2]:
# values that are given in the problem

fluidA = 'Ammonia'
fluidB = 'CarbonDioxide'

T1 = -20+273.15 # Freezer operating at -20C
T3 = 273.15-12#[i+273.15 for i in range(-10,11)]    #range of T_intermediate/Temp. heat exchanger to look for best COP
T5 = T3

T_ambient = 20 + 273.15  # Ambient temperature in Kelvin of average East Bay Day
T_approach = -10  # Approach temperature
T_condensation = T_ambient + T_approach  # Condenser temperature in Kelvin

# Saturation pressure of ammonia at T_condensation
T7 = T_condensation
print(T7)

283.15


In [3]:
# state 1
Q1 = 1 # quality
h1 = PropsSI('H','T',T1,'Q',Q1,fluidB)
s1 = PropsSI('S','T',T1,'Q',Q1,fluidB)
P1 = PropsSI('P','T',T1,'Q',Q1,fluidB)
print('Pressure of frezzer is: {0:1.2f}, which is within the normal range'.format(P1))
print(T1)

Pressure of frezzer is: 1969628.00, which is within the normal range
253.14999999999998


In [4]:
# state 3
Q3 = 0
h3 = PropsSI('H','T',T3,'Q',Q3,fluidB)
s3 = PropsSI('S','T',T3,'Q',Q3,fluidB)
P3 = PropsSI('P','T',T3,'Q',Q3,fluidB)
print(P3)
print(T3)

2500952.2835903815
261.15


In [5]:
# state 2
s2 = s1 # isentropic compressor
P2 = P3 # isobaric
h2 = PropsSI('H','P',P2,'S',s2,fluidB)
s2 = PropsSI('S','P',P2,'S',s2,fluidB)
T2 = PropsSI('T','P',P2,'S',s2,fluidB)
print(P2)
print(T2)

2500952.2835903815
268.83536975765173


In [6]:
# state 4
P4 = P1 # isobaric
h4 = h3 # isenthalpic
s4 = PropsSI('S','P',P4,'H',h4,fluidB)
T4 = PropsSI('T','P',P4,'H',h4,fluidB)
Q4 = PropsSI('Q','P',P4,'H',h4,fluidB)
print(P4)
print(T4)

1969628.0019025588
253.15000000000822


## High-Temperature Cycle (Ammonia Cycle)

**5-6 Isentropic Compression:** Ammonia in vapor form is compressed to a higher pressure, increasing its temperature. The compression process prepares the refrigerant for heat rejection in the condenser.

**6-7 Isobaric Heat Rejection:** The high-pressure, high-temperature ammonia vapor releases heat to the environment (e.g., air or water) in the condenser. This step condenses the refrigerant into a high-pressure liquid.

**7-8 Isenthalpic Expansion:**  The high-pressure ammonia liquid passes through an expansion valve, where its pressure and temperature drop significantly. This prepares the refrigerant for heat absorption in the cascade heat exchanger.

**8-5 Isobaric Heat Absorption:** The low-pressure, low-temperature ammonia refrigerant absorbs heat from the CO2 refrigerant in the cascade heat exchanger. This step ensures the ammonia cycle removes the heat from the low-temperature CO2 cycle and maintains the intermediate temperature.

Here are the properties for cycle A, Ammonia cycle:

In [7]:
# state 5
Q5 = 1 # saturated vapor 
h5 = PropsSI('H','T',T5,'Q',Q5,fluidA)
s5 = PropsSI('S','T',T5,'Q',Q5,fluidA)
P5 = PropsSI('P','T',T5,'Q',Q5,fluidA)
print(P5)
print(T5)

267747.80001646694
261.15


In [8]:
# state 7
Q7 = 0 # vapor quality at the exit of the condenser is 0
h7 = PropsSI('H','T',T7,'Q',Q7,fluidA)
s7 = PropsSI('S','T',T7,'Q',Q7,fluidA)
P7 = PropsSI('P','T',T7,'Q',Q7,fluidA)
print(P7)
print(T7)

614790.2086291318
283.15


In [9]:
# state 6
s6 = s5 # compressor operates isentropically 
P6 = P7 #Isobaric condensation
T6 = PropsSI('T','P',P6,'S',s6,fluidA)
h6 = PropsSI('H','P',P6,'S',s6,fluidA)
s6 = PropsSI('S','P',P6,'S',s6,fluidA)
print(P6)
print(T6)

614790.2086291318
316.0326977750857


In [10]:
# state 8
h8 = h7 # isenthalpic process going from state 7 to 8
P8 = P5 # isobaric process going from state 8 to state 5
s8 = PropsSI('S','P',P8,'H',h8,fluidA)
T8 = PropsSI('T','P',P8,'H',h8,fluidA)
Q8 = PropsSI('Q','P',P8,'H',h8,fluidA)
print(P8)
print(T8)

267747.80001646694
261.1499999999947


#### Calculating Work based on Different Intermediate Temperatures of the Heat Exchanger

Let's assume $$\implies \dot{m}_\text{amm} = 0.05 kg/s$$

In [11]:
mdotAmm_kgPerS = 0.05

Lets calculate the Qin for ammonia cycle, Win of ammonia cycle, and Qout from ammonia cycle to environment.

In [12]:
Qamm_in = mdotAmm_kgPerS*(h5 - h8)
print('The rate of heat removal into ammonia cycle is {0:1.2f} W.'.format(Qamm_in))

Wamm_in = mdotAmm_kgPerS*(h6 - h5)
print('The power input to the compressor in ammonia cycle is {0:1.2f} W.'.format(Wamm_in))

Qamm_out = mdotAmm_kgPerS*(h6 - h7)
print('The heat rejected from ammonia cycle to the environment is {0:1.2f} W.'.format(Qamm_out))

The rate of heat removal into ammonia cycle is 60065.60 W.
The power input to the compressor in ammonia cycle is 5537.86 W.
The heat rejected from ammonia cycle to the environment is 65603.46 W.


Due to energy balance, we know that the heat out of the CO2 cycle must equal to the heat in of the ammonia cycle. This leads to $$\dot{Q}_\text{amm,in} =  \dot{Q}_\text{CO2,out}$$

In [13]:
QCO2_out = -Qamm_in

Since we know that $$\dot{Q}_\text{CO2,out} = \dot{m} \left( h_3 - h_2 \right) $$

$$\implies  \dot{m}_\text{CO2} = \frac{\dot{Q}_\text{CO2,out}}{\left( h_3 - h_2 \right)}$$

In [14]:
mdotCO2_kgPerS = QCO2_out/(h3 - h2)
print('The mass flow rate in CO2 cycle is {0:1.3f} kg/s.'.format(mdotCO2_kgPerS))

The mass flow rate in CO2 cycle is 0.219 kg/s.


Now lets calculate the Qin of CO2 cycle, or heat removed from freezer, and also the Win total by finding WCO2_in and adding it to Wamm_in.
$$\dot{W}_\text{in} = \dot{m}_\text{amm}\left(h_6 - h_6 \right) + \dot{m}_\text{CO2}\left(h_2 - h_1\right)$$.

In [15]:
QCO2_in = mdotCO2_kgPerS *(h1 - h4)
print('The rate of heat removal from states 4 to 1 is {0:1.3f} W.'.format(QCO2_in))

WCO2_in = mdotCO2_kgPerS * (h2 - h1)
WdotIn_W = Wamm_in + WCO2_in
print('The total power input to the compressors is {0:1.3f} W.'.format(WdotIn_W))

The rate of heat removal from states 4 to 1 is 58020.669 W.
The total power input to the compressors is 7582.788 W.


The coefficient of performance of any refrigeration system is the ratio of the rate that heat is removed from the refrigated space divided by the net power input:

$$\text{COP}_\text{R} = \frac{\dot{Q}_\text{CO2,out}}{\dot{W}_\text{in}} $$


In [16]:
COP_R = QCO2_in/WdotIn_W
print('The COP of the cascade refrigeration cycle is {0:1.3f}.'.format(COP_R))

The COP of the cascade refrigeration cycle is 7.652.


Now Iterate this process for different intermediate temperature (T3) to find the highest COP_R possible. Testing a range of (-15,5)C for because that is the normal range for commerical cascade refrigeration system. Here is a table of the tested T_intermediate values with their COPs. 

We also test -20 and 10 to make sure the trend of COP is stable.


| T_intermediate       | COP       |
|----------------|----------------|
| -20   | 7.454   |
| -15   | 7.624   |
| -14   | 7.64   |
| -13   | 7.649   |
| <mark>-12</mark>   | <mark>7.652</mark>   |
| -11   | 7.648   |
| -10   | 7.637   |
| -9   | 7.619   |
| -8   | 7.595   |
| -7  | 7.564  |
| -6  | 7.526  |
| -5  | 7.482  |
| -4  | 7.431  |
| -3  | 7.374  |
| -2  | 7.31  |
| -1  | 7.241  |
| 0  | 7.165  |
| 1  | 7.084  |
| 2  | 6.997  |
| 3  | 6.905  |
| 4  | 6.808  |
| 5  | 6.706  |
| 10  | 6.132  |

We find that around -12 degree Celcius is the optimal temperature for CO2 in the condenser and ammonia in the evaporator installing this system in the East bay (T_ambient ~20C).

-----

## Improving the performance

One way to improve the performance is to replace the expansion valves that is used for the isenthaplic expansion from states 3 to 4 and 7 to 8 with turbines (isentropic expansion)
that can convert some of the energy lost in the expansion process into useful work. This means we have to recalculate state 4 and 8 and also the rest of the W and Q.

In [17]:
# new state 4
P4 = P1 # isobaric
s4 = s3 # isentropic
h4 = PropsSI('H','P',P4,'S',s4,fluidB)
T4 = PropsSI('T','P',P4,'S',s4,fluidB)
Q4 = PropsSI('Q','P',P4,'S',s4,fluidB)
print(P4)
print(T4)

# state 8
s8 = s7 # isentriouc process going from state 7 to 8
P8 = P5 # isobaric process going from state 8 to state 5
h8 = PropsSI('H','P',P8,'S',s8,fluidA)
T8 = PropsSI('T','P',P8,'S',s8,fluidA)
Q8 = PropsSI('Q','P',P8,'S',s8,fluidA)
print(P8)
print(T8)

1969628.0019025588
253.15000000000822
267747.80001646694
261.1499999999947


Now calculate everything to get new COP. Work outputed from the turbine will offset compressor work directly by mechanical coupling.

In [18]:
mdotAmm_kgPerS = 0.05

Qamm_in = mdotAmm_kgPerS * (h5 - h8)
#print('The rate of heat removal into ammonia cycle is {0:1.2f} W.'.format(Qamm_in))

Wamm_in = mdotAmm_kgPerS * (h6 - h5)
#print('The power input to the compressor in ammonia cycle is {0:1.2f} W.'.format(Wamm_in))

Qamm_out = mdotAmm_kgPerS * (h6 - h7)
#print('The heat rejected from ammonia cycle to the environment is {0:1.2f} W.'.format(Qamm_out))

QCO2_out = -Qamm_in

mdotCO2_kgPerS = QCO2_out/(h3 - h2)
#print('The mass flow rate in CO2 cycle is {0:1.3f} kg/s.'.format(mdotCO2_kgPerS))

QCO2_in = mdotCO2_kgPerS * (h1 - h4)
print('The rate of heat removal from states 4 to 1 is {0:1.3f} W.'.format(QCO2_in))

WCO2_in = mdotCO2_kgPerS * (h2 - h1)

WCO2_out = mdotCO2_kgPerS * (h3 - h4)
WAMM_out = mdotAmm_kgPerS * (h7 - h8)
print('The total power output from the CO2 cycle turbine is {0:1.3f} W.'.format(WCO2_out))
print('The total power output from the Ammonia cycle turbine is {0:1.3f} W.'.format(WAMM_out))

WdotIn_W = Wamm_in + WCO2_in - WCO2_out - WAMM_out
print('The total power input to the compressors is {0:1.3f} W.'.format(WdotIn_W))

COP_R = QCO2_in/WdotIn_W
print('The COP of the cascade refrigeration cycle is {0:1.3f}.'.format(COP_R))

The rate of heat removal from states 4 to 1 is 58416.040 W.
The total power output from the CO2 cycle turbine is 173.753 W.
The total power output from the Ammonia cycle turbine is 229.430 W.
The total power input to the compressors is 7187.416 W.
The COP of the cascade refrigeration cycle is 8.128.


----

## Conclusions

We can see that the turbine outputs around 200 W of work each which is fed back into the compressors. This decreases the total power input to the compressor needed and increase the COP from 7.652 to 8.128 for T=-12C. We can also see that the cooling capacity of the system is at around 58kW. A  cooling load of 58 kW is typically enough to:
- Freeze or chill approximately 10-15 tons of food per day (depending on the food type and freezing requirements).
- Maintain a small-to-medium-sized industrial freezer room at -20C.
- Operate a small-to-medium cryogenic system for specialized applications.
- Cooling for small-to-medium-sized data centers where heat dissipation is critical for server performance and reliability.